src/components/Modal/BaseModal.js

```diff
@@ -16,11 +16,15 @@ const propTypes = {
 
     /** The ref to the modal container */
     forwardedRef: PropTypes.func,
+
+    /** Ensure that callback and trap deactivation are in the same loop on the web platform */
+    useOnDismiss: PropTypes.bool,
```

Comment:
Could you please rename this to follow our naming pattern for boolean props? It should be `shouldUserOnDismiss`.. We usually prefix booleans with them with `is` or `should`